Source/Projections/Store/CurrentStateType.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Dolittle.SDK.Projections.Store
+{
+    /// <summary>
+    /// Defines the different types a projection state can have.
+    /// </summary>
+    public enum CurrentStateType
```

Comment:
We should probably also use this in the `ProjectionContext` somehow - that code is slightly hacky right now. I think it is located in the event processor for a projection I think.

Reply:
I don't really understand what you want here? To have a property with this type in the ProjectionContext?

Reply:
Not really. This CurrentStateType feels a bit "internal" to the protocol. And right now, it's just describing if the state you got back for the projection was stored - or if you got the default values. In the `ProjectionContext` we "reduced" this enum into a bool saying `WasCreatedFromInitialState`.

Reply:
Ahh. So removing the type?

Reply:
As in this type

Reply:
I'll fix this in a later PR. In the projection processor - we skipped the conversion to a Runtime representation of this protobuf enum. So it's just about making it nicer. But I'll do it later :)

Reply:
👍